src/cache/CacheManager.ts

```diff
@@ -56,58 +50,27 @@ export class CacheManager {
+        const coordsInMemory: ILocationDictionary = this.memoryCache.loadCoordinates(locations); // {"London": {"lat": 54, "lon": 34"}, "Moscow": {"lat": 64, "lon": 54"}
         locationsInMemory = Object.keys(coordsInMemory);
-        
         locations = locations.filter(loc => !locationsInMemory.includes(loc));                        
-        
         if (locations.length === 0) {
             result = Object.assign({}, coordsInMemory);
             return result;
         }
 
-
-        // Load from localStorage
-        if (isEmpty(this.coordsInLocalStorage)) {
-            return this.createLocalStorageCache()
-                .then(cache => cache.loadCoordinates(locations))
-                .then(async (coordinatesPromise: Promise<ILocationDictionary>) => {
-                    const coordinates = await coordinatesPromise;
-                    if (coordinates && Object.keys(coordinates).length > 0) {
-                        if (isEmpty(this.coordsInLocalStorage)) {
-                            this.coordsInLocalStorage = coordinates;
-                        }
-                        
-                        if (this.coordsInLocalStorage) {
-                            const locationsInLocalStorage = Object.keys(this.coordsInLocalStorage);
-                            locations = locations.filter(loc => !locationsInLocalStorage.includes(loc));
-                    
-                            if (locations.length === 0) {
-                                result = Object.assign({}, locationsInMemory, this.coordsInLocalStorage);
-                                return result;
-                            }
-    
-                            // Load additional locations from Bing
-                            const locationsFromBing = await this.getLocationsFromBing(locations, locationsInMemory, locationsDictionary);
-                            return locationsFromBing;
-                        }
-                    }
-                    else {
-                        console.log("Local storage is empty, will attempt to load the coordinates from Bing API");
-                        
-                        const locationsFromBing = await this.getLocationsFromBing(locations, locationsInMemory, locationsDictionary);
-                        return locationsFromBing;
-                    }
-                    
-                }).catch(async (e) => {
-                    console.error("Error while loading coordinates from local storage", e);
-                    const locationsFromBing = await this.getLocationsFromBing(locations, locationsInMemory, locationsDictionary);
-                    return locationsFromBing;
-            });
-        }
-        else {
-            return this.coordsInLocalStorage;
+        // Load from local storage
+        const coordsInLocalStorage: ILocationDictionary = await this.createLocalStorageCache().then(cache => cache.loadCoordinates(locations));
```

Comment:
we don't need a new instance of localstorage cache each time visual needs new coordinates.